Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+        {
+            bool returnAllPlatforms = ProjectType == ProjectType.PredefinedAssembly
+                || (inEditor && ProjectType == ProjectType.PredefinedEditorAssembly)
+                || (inEditor && ProjectType == ProjectType.EditorAsmDef)
+                || (inEditor && ProjectType == ProjectType.AsmDef && AssemblyDefinitionInfo.EditorPlatformSupported);
+
+            if (returnAllPlatforms)
+            {
+                return new ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo>(availablePlatforms.ToDictionary(t => t.BuildTarget, t => t));
+            }
+
+            bool returnNoPlatforms = (!inEditor && ProjectType == ProjectType.PredefinedEditorAssembly)
+                || (!inEditor && ProjectType == ProjectType.EditorAsmDef)
+                || (!inEditor && ProjectType == ProjectType.AsmDef && AssemblyDefinitionInfo.TestAssembly);
+
+            if (returnNoPlatforms)
+            {
+                return new ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo>(new Dictionary<BuildTarget, CompilationPlatformInfo>());
+            }
+
+            // We only are an asmdef at this point, as we handle all predefined assembly at line 130, inEditor && predefined editor assembly at line 130, and !inEditor && predefined editor assembly at line 139
+            Func<CompilationPlatformInfo, bool> predicate = AssemblyDefinitionInfo.includePlatforms.Length > 0
+                ? predicate = (t) => AssemblyDefinitionInfo.includePlatforms.Contains(t.Name)
+                : predicate = (t) => !AssemblyDefinitionInfo.excludePlatforms.Contains(t.Name);
+
+            return new ReadOnlyDictionary<BuildTarget, CompilationPlatformInfo>(
+                availablePlatforms.Where(predicate)
+                    .ToDictionary(t => t.BuildTarget, t => t));
+        }
+
+        /// <summary>
+        /// Adds a dependency to the project.
+        /// </summary>
+        /// <param name="csProjectInfo">The C# dependency.</param>
+        internal void AddDependency(CSProjectInfo csProjectInfo)
+        {
+            AddDependency(csProjectDependencies, csProjectInfo);
+        }
+
+        /// <summary>
+        /// Adds a dependency to the project.
+        /// </summary>
+        /// <param name="pluginAssemblyInfo">The plugin dependency.</param>
+        internal void AddDependency(PluginAssemblyInfo pluginAssemblyInfo)
+        {
+            AddDependency(pluginAssemblyDependencies, pluginAssemblyInfo);
+        }
+
+        private void AddDependency<T>(List<CSProjectDependency<T>> items, T referenceInfo) where T : ReferenceItemInfo
+        {
+            items.Add(new CSProjectDependency<T>(referenceInfo,
+                new HashSet<BuildTarget>(InEditorPlatforms.Keys.Intersect(referenceInfo.InEditorPlatforms.Keys)),
+                new HashSet<BuildTarget>(PlayerPlatforms.Keys.Intersect(referenceInfo.PlayerPlatforms.Keys))));
```

Comment:
This takes an intersection here because our new dependency can only support the subset that both of these references support (i.e. if we support A, B, this other thing supports B, C, then we can only really support B between the two of us) right?

Reply:
Yes

---
In reply to: [301798052](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301798052) [](ancestors = 301798052)